flink/src/main/java/org/apache/iceberg/flink/CatalogLoader.java

```diff
@@ -94,4 +100,47 @@ public String toString() {
           .toString();
     }
   }
+
+  class CustomCatalogLoader implements CatalogLoader {
+
+    private final SerializableConfiguration hadoopConf;
+    private final String name;
+    private final String impl;
+
+    private CustomCatalogLoader(String name, Configuration conf, String impl) {
+      this.hadoopConf = new SerializableConfiguration(conf);
+      this.name = name;
+      this.impl = Preconditions.checkNotNull(impl,
+          "Cannot initialize custom Catalog because impl property is not set");
+    }
+
+    @Override
+    public Catalog loadCatalog() {
```

Comment:
Maybe this method can be static and be called from other places.

Reply:
It implements the `CatalogLoader` interface and cannot be changed to static.

Reply:
What about using a static method in some other class in core? Maybe `CatalogUtil.loadCatalog`?

Reply:
Yeah sounds like a good idea to move the logic to a common util, let me do that